### PR TITLE
add - collapsed menu with one extra-btn

### PR DESCRIPTION
**Pull request by lanwen**
## Collapsed menu
Can save up to 5% of screen if no need to click on tabs every time
  

Hilight on hover

Left align

### CODE REVIEW

**Comment by just-boris** [comment]
What the reason of this?

**Comment by lanwen** [comment]
More content-oriented behaviour. U use nav extremely less than view content

**Comment by arpitgold** [comment]
:+1: Nice and Trending Feature

**Comment by vania-pooh** [comment]
@lanwen I would somehow save whether collapsed state is activated and restore it on page reload.

**Comment by baev** [comment]
@lanwen thanks for sending us the pull request!

I personally more like always small icons. And we can show bigger icon + tab name by mouse hover

**Comment by lanwen** [comment]
@baev No, mouse_over event is bad idea - it will be same as epileptic

**Comment by just-boris** [comment]
So, I don't see significant difference.

Here is two screenshots of whole page with the same scale:

image

image

You will get about 40px, 4% from 1024px. In other words you will see 3 extra characters in line.
It is ridiculous, this feature doesn't worth so much.

**Comment by just-boris** [comment]
Finally, the most nice improvement for sidebar would be to keep it aligned with logo.
image
It could be make report better anyway

**Comment by lanwen** [comment]
@just-boris so, maybe just scale logo?

**Comment by lanwen** [comment]
@just-boris it's not ridiculous, ridiculous to show big menu every time with lots of words, spaces when you only need to view tests. Menu is a service stuff, and should not be so bright and big

**Comment by eroshenkoam** [comment]
retest this please

**Comment by lanwen** [comment]
one more change will be: small icons as default

**Comment by lanwen** [comment]
ill squash commits

**Comment by lanwen** [comment]
rdy for last review and merge

**Comment by baev** [comment]
LGTM!

**Comment by baev** [comment]
could you please remove the shadow 



Also seems like better to place icon on the center

**Comment by just-boris** [comment]
I got a collapsed menu by default. Please, make it expanded at first time

File: allure-report-face/src/main/webapp/js/controllers.js [hunk 1]
```diff
@@ -74,8 +74,22 @@ angular.module('allure.controllers', [])
...
+            return settings.getItem('collapsed');
+        };
+
+        $scope.toggleCollapsed = function() {
+            settings.setItem('collapsed', !$scope.isCollapsed());
+        };
+
+        if($scope.isCollapsed() == undefined) {
+            $scope.toggleCollapsed();
+        }
```
Comment (just-boris) [inline allure-report-face/src/main/webapp/js/controllers.js]: I am not sure, that this `if` needed. We will save this setting after the first click on button

Comment (lanwen) [inline allure-report-face/src/main/webapp/js/controllers.js]: we need a collapsed state by default. I don't know how init it without this `lf` statement

**Comment by lanwen** [comment]
collapsed by default is expected behaviour

**Comment by baev** [comment]
@lanwen when you first time look the report better if you can see tab names => default is uncollapsed

**Comment by lanwen** [comment]
don't think so. Bitbucket is example

**Comment by just-boris** [comment]
We tried to create more nice version of button
Here is result


That is better, because doesn't have a lot of borders and inherits more styles from sidebar icons.
Also we are used font-awesome icon instead of custom one

**Comment by lanwen** [comment]
retest this please

**Comment by lanwen** [comment]
retest this please

**Comment by eroshenkoam** [comment]
retest this please

**Comment by lanwen** [comment]
test this please

**Comment by eroshenkoam** [comment]
test this please

**Comment by baev** [comment]
lol

**Comment by lanwen** [comment]
Jenkins ignores us

**Comment by baev** [comment]
Jenkins CI : Rise of the Machines

**Comment by lanwen** [comment]
upi! Thx!

**Comment by baev** [comment]
@lanwen GJ, thanks! :+1:

**Pull request merged**